File: Python/cash.py
from cs50 import get_float

def main():

    # variable to keeps track of amount of coins
    coins = 0

    # get positive float from user
    i = get_pos_float("Change Owed: ")

    # change the float to a whole number
    change = i * 100

    # get the amount of quarters
    while change >= 25:
        change -= 25
        coins += 1

    # get the amount of dimes
    while change >= 10:
        change -= 10
        coins += 1

    # nickels...
    while change >= 5:
        change -= 5
        coins += 1

    # and pennies...
    while change >= 1:
        change -= 1
        coins += 1

    # print the amount of coins it took to creat the change
    print(coins)


 # function to keep asking for positive float from user as long as it's not positive
def get_pos_float(prompt):
    while True:
        num = get_float(prompt)
        if num > 0:
            break
    return num

if __name__ == "__main__":
    main()

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float amount;
    int coins = 0; //variable to keep track of each coin used

    // make sure the input is a float
     do {
        amount = get_float("Change owed: "); // get the amount of change
    } while (amount < 0);

    // convert the float input to an whole number integers
    // easier to work with
    int change = (int) round(amount * 100);
    // get the amount of quarters
    int quarters = change / 25;
    int dimes = (change % 25) / 10;
    int nickels = ((change % 25) % 10) / 5;
    int pennies = ((change % 25) % 10) % 5;

    int change_owed = quarters + dimes + nickels + pennies;

    printf("%i\n", change_owed);

}
